code/obj/item/flamethrower.dm

```diff
@@ -79,6 +79,20 @@ A Flamethrower in various states of assembly
 		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
 			return TRUE
 
+
+	/// allow refilling the fuel tank by simply clicking the reagent dispensers
+	afterattack(atom/target, mob/user, flag)
+		if(istype(target, /obj/reagent_dispensers) && in_interact_range(src,target))
```

Comment:
lol you could refill from ants on the floor

Reply:
lol